api/views.py

```diff
@@ -0,0 +1,112 @@
+from rest_framework import permissions
+from django.http import Http404
+from rest_framework.views import APIView
+from rest_framework.response import Response
+from rest_framework import exceptions
+from .api_exceptions import ServiceUnavalibleException
+import requests
+
+
+class RawQuery(APIView):
+    """
+    Acts as a router to the db.sead.systems:8080 api, nothing more.
+    you can pass in any of the supported query parameters
+    ('type', 'start_time', 'end_time', 'device', 'diff', 'subset', 'limit', 'json')
+    """
+    permission_classes = []
+    allowed_parameters = ['type', 'start_time', 'end_time', 'device',
+                      'diff', 'subset', 'limit', 'json']
+
+    def get(self, request, device_id):
+        """
+        :summary: This is the get request for the raw_query \n
+        :param request: the object containing the query parameters \n
+        :param device_id: the device_id of the device requesting information about\n
+        :return: \n
+        """
+        params = request.GET.dict()
+
+        for param in params:
+            if param not in self.allowed_parameters:
+                raise ValueError('Query parameter not allowed: ', param)
+
+        response = requests.get("http://db.sead.systems:8080/" + device_id, params=params)
+        response.raise_for_status()
+
+        return Response({"status": response.status_code , "data": response.json()})
+
+
+class TotalPower(APIView):
+
+    """ Calculates the total power usage or generation for a particular device over some time period
+    include start_time and end_time as query paramers in api call. Start and end times
+    must be utc unix timestamps. """
+
+    permission_classes = []
+    allowed_parameters = ['start_time', 'end_time']
+    devices = ['Panel1', 'Panel2', 'Panel3', 'shed ', 'PowerS', 'PowerG']
+
+    def get(self, request, device_id, power_type):
+
+        """
+        :summary: Get request for total power\n
```

Comment:
I just wanted nice looking api documentation. honestly a couple of '\n' in the code does not bother me if it formats the documentation nicely. I can submit a bug to swagger about this problem but in the meantime id rather the documentation be as clear as possible so that my team meets can understand it quickly.